### PR TITLE
fixed a bad-am_obj-use bug

### LINKED ISSUE #376: Decompilation view crashes when tabbing back to disassembly
```Traceback (most recent call last):
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/ui/widgets/qccode_edit.py", line 145, in event
    return self.keyPressEvent(event)
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/ui/widgets/qccode_edit.py", line 194, in keyPressEvent
    if self._code_view.keyPressEvent(event):
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/ui/views/code_view.py", line 253, in keyPressEvent
    self.workspace.jump_to(self.addr.am_obj)
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/ui/workspace.py", line 261, in jump_to
    view.jump_to(addr, use_animation=use_animation)
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/ui/views/disassembly_view.py", line 522, in jump_to
    self._jump_to(addr, use_animation=use_animation)
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/ui/views/disassembly_view.py", line 687, in _jump_to
    self.infodock.select_instruction(instr_addr, unique=True, use_animation=use_animation)
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/logic/disassembly/info_dock.py", line 119, in select_instruction
    self.selected_insns.am_event(insn_addr=insn_addr)
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/data/object_container.py", line 25, in am_event
    listener(**kwargs)
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/plugins/trace_viewer/qtrace_viewer.py", line 299, in _on_select_ins
    positions = self.trace.get_positions(addr)
  File "/Users/mahaloz/github/angr-dev/angr-management/angrmanagement/data/object_container.py", line 60, in __getattr__
    return getattr(self._am_obj, item)
AttributeError: 'NoneType' object has no attribute 'get_positions'
```

Binary is one we use for testing, attached. 
[basic_buffer_overflows.zip](https://github.com/angr/angr-management/files/6771300/basic_buffer_overflows.zip)

### PR DESCRIPTION
**Pull request by mahaloz**
Fixed a bug in TraceViewer bug.

Closes #376.

### CODE REVIEW

**Comment by mahaloz** [comment]
@rhelmot the AMContainer stuff is as dangerous as it is useful, since all plugins (loaded or not), seem to be able to crash if they handle publish events from an AMContainer and something goes wrong. Ideally, we should probably find a way to either isolate plugins more.

**Comment by ltfish** [comment]
I think the solution is to add `try: ... except ...` around all callbacks to plugins.

**Review by ltfish — APPROVED** [review]

File: angrmanagement/plugins/trace_viewer/qtrace_viewer.py [hunk 1]
```diff
@@ -282,7 +282,7 @@ def _on_tab_change(self):
             self._show_trace_ids()
 
     def _on_select_ins(self, **kwargs): # pylint: disable=unused-argument
-        if self.trace is None:
+        if self.trace.am_obj is None:
```
Comment (ltfish) [inline angrmanagement/plugins/trace_viewer/qtrace_viewer.py:285]: Use `if self.trace.am_none:` instead.

**Review by ltfish — CHANGES_REQUESTED** [review]

**Review by ltfish — APPROVED** [review]

**Comment by rhelmot** [comment]
> I think the solution is to add try: ... except ... around all callbacks to plugins.

I architected the plugin manager so that it did its own error handling - did we add any additional callbacks that didn't use the `PluginManager._dispatch` function?

**Comment by ltfish** [comment]
I don't know. Maybe?

**Comment by rhelmot** [comment]
Oh, I misunderstood the problem. You're right, we should absolutely add try-except inside am_event.

**Comment by rhelmot** [comment]
See linked commit. Should prevent future issues from being catastrophic.

**Comment by ltfish** [comment]
Python 3.11 implements zero-cost exception handling. Performance-wise, adding try-except inside am_event() should be good in the long run.

**Comment by ltfish** [comment]
@rhelmot Can we print the entire stack trace for the ease of debugging?

**Comment by rhelmot** [comment]
Currently it looks like this:

```
ERROR   | 2021-07-06 21:20:15,639 | angrmanagement.data.object_container | Error raised from event of (container: Global states list)[<SimState @ 0x4004b0>]
Traceback (most recent call last):
  File "/home/audrey/angr/angr-management/angrmanagement/data/object_container.py", line 26, in am_event
    listener(**kwargs)
  File "<ipython-input-3-1889d8869666>", line 2, in myfunc
    raise Exception('haha')
Exception: haha
```

I don't know how to get python to display a longer backtrace than that. It does print out the exact object container that was used, which should be helpful for debugging...

**Comment by ltfish** [comment]
`l.exception(..., exc_info=True)`?

**Comment by rhelmot** [comment]
`exception` has `exc_info=True` by default. that's what distinguishes it from `error`.

```
[+] In [3]: l.exception??
Signature: l.exception(msg, *args, exc_info=True, **kwargs)
Source:   
    def exception(self, msg, *args, exc_info=True, **kwargs):
        """
        Convenience method for logging an ERROR with exception information.
        """
        self.error(msg, *args, exc_info=exc_info, **kwargs)
File:      /usr/lib/python3.8/logging/__init__.py
Type:      method
```

**Pull request merged**